core/src/main/java/com/nisovin/magicspells/spells/PlayerMenuSpell.java

```diff
@@ -176,16 +209,18 @@ private void processClickSpell(Subspell subspell, Player caster, Player target,
 		subspell.cast(caster, power);
 	}
 
-	private void open(Player opener, SpellData data) {
+	private void open(Player opener, MenuData data, int page) {
+		SpellData spellData = data.spellData();
+
 		List<Player> players = new ArrayList<>(Bukkit.getOnlinePlayers());
 		if (!addOpener) players.remove(opener);
 		if (playerModifiers != null) players.removeIf(player -> !playerModifiers.check(player));
 		if (radius > 0) players.removeIf(player -> opener.getLocation().distance(player.getLocation()) > radius);
 
-		int size = (int) Math.ceil((players.size()+1) / 9.0) * 9;
+		int size = (int) Math.ceil(Math.min((players.size()+1), 54) / 9.0) * 9;
 		Inventory inv = Bukkit.createInventory(opener, size, Component.text(internalName));
 
-		for (int i = 0; i < players.size(); i++) {
+		for (int i = (page * 50); i < Math.min(players.size(), (page + 1) * 50); i++) {
```

Comment:
The amount of players per page should probably be `52`, not `50`, to avoid unused space.